src/ca/legacy/pcas/generic/casStrmClient.cc

```diff
@@ -1487,7 +1487,7 @@ caStatus casStrmClient::hostNameAction ( epicsGuard < casClientMutex > & guard )
         }
         return S_cas_internal;
     }
-    strncpy ( pMalloc, pName, size - 1 );
+    memcpy ( pMalloc, pName, (size - 1)*sizeof(char));
     pMalloc[ size - 1 ]='\0';
```

Comment:
What warning is coming from this instance?  This looks like as "safe" a string duplication as I can imagine.

```c++
size = strlen(pName)+1u;
pMalloc = new char [size];
if(!pMalloc) { ... }
strncpy ( pMalloc, pName, size - 1 );
pMalloc[ size - 1 ]='\0';
```

The bug I see here is expecting c++ `new[]` to return `NULL` when it will actually throw an exception.

Reply:
imo. plain `strcpy()` here would be appropriate since local code has just allocated the destination buffer.

Reply:
This warning

```
../../../../../../src/ca/legacy/pcas/generic/casStrmClient.cc:1490:13: warning: ‘char* strncpy(char*, const char*, size_t)’ output truncated before terminating nul copying as many bytes from a string as its length [-Wstringop-truncation]                  
     strncpy ( pMalloc, pName, size - 1 );                                                                                                                                                                                                                     
     ~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~                                                                                                                                                                                                                      

```

I can remove this kind of warning using this:

```
#pragma GCC diagnostic push
#pragma GCC diagnostic ignored "-Wstringop-truncation"
    strncpy ( pMalloc, pName, size - 1 );
#pragma GCC diagnostic pop
```

As this is not really an issue

Reply:
I don't like disabling whole classes of warnings now and "forever".  Better to replace with plain `strcpy()` since there is no danger of truncation or overflow in this instance.

Reply:
Would a non gcc compiler generate an "unknown pragma" warning?

Reply:
@mdavidsaver , but this would be applied only for one command specific, not for all the code.  But if you still think is better to replace by strcpy I can do that.

Reply:
@FreddieAkeroyd , not sure, I will check.

Reply:
Also, I think that GCC is specifically complaining that both of the following lines _could_ write to `pMalloc[size-1]`.

```c
strncpy ( pMalloc, pName, size - 1 );
pMalloc[ size - 1 ]='\0';
```

So the warning would also go away by changing the `strncpy()` to `size-2`, although that would also require handing the `size==1` case specially.

Reply:
You are right, it is possible to solve this warning this way.

Reply:
> ... would be applied only for one command specific, not for all the code.

If you can find a one-size-fits-all solution for `char*` operations in C, then you will be famous!  I think these warnings need to be considered case by case.

Reply:
Yes, you are right. I will start this again, looking case by case.